contracts/eosio.system/src/eosio.system.cpp

```diff
@@ -397,6 +397,59 @@ namespace eosiosystem {
       set_resource_limits( account, current_ram, current_net, cpu );
    }
 
+   void system_contract::denynames( const std::vector<name>& patterns ) {
+      if (patterns.empty())
+         return; // no-op for empty list, consistent with ignoring duplicate names.
+
+      require_auth( get_self() );
```

Comment:
I think the `require_auth` should still come first. Even though calling `denynames` with no patterns is a no-op, I rather not allow people add it to the blockchain without the appropriate permissions.